### DISCUSSION TITLE
Function stripe-webhook Relative import path "http" not prefixed with

### DISCUSSION THREAD

**Discussion by louis030195** [supabase/supabase#13374] [Questions] [answered]
Hi, I'm trying to run this example:

https://github.com/supabase/supabase/blob/master/examples/edge-functions/supabase/functions/stripe-webhooks/README.md

It's my first time using Deno, installed everything but I face this issue:

```
error: Uncaught (in promise) Error: Relative import path "http" not prefixed with / or ./ or ../ and not in import map from "https://esm.sh/v113/@types/node@16.18.12/http.d.ts"
      const ret = new Error(getStringFromWasm0(arg0, arg1));
```

When running

```bash
supabase functions deploy --no-verify-jwt stripe-webhook
```

Also, wasn't able to get any understanding from the local testing, basically nothing happen
I at least expected that the console log would show but nothing?

Ran

```bash
supabase functions serve --no-verify-jwt --env-file ./supabase/.env.local
```

With my .env.local filled with both Stripe credentials
Then

```bash
stripe listen --forward-to localhost:54321/functions/v1/
```

then

```bash
stripe trigger payment_intent.succeeded
```

I get this in first stripe terminal:

<img width="728" alt="image" src="https://user-images.githubusercontent.com/25003283/229350991-7f894c2c-ece8-4c9d-aee5-b1976ff8b37a.png">

And nothing in supabase serve terminal

(previously tried to find similar issue and I asked GPT-4 for a solution without any success :( )

**Comment by deepak-coding-art** [comment]
I am having same problem 
Did you find any solution

**Reply by louis030195** [reply]
no
both this and https://github.com/vercel/nextjs-subscription-payments fails (https://github.com/vercel/nextjs-subscription-payments/issues/176)
I'll switch to google cloud

**Reply by tiniscule** [reply]
I believe this might be because support for multiple functions is now live on local dev.  You may just need to add the name of your function to the URL.  for example `http://localhost:5432/functions/v1/stripe-webhooks` if that's what it's called.

**Comment by deepak-coding-art** [comment, marked as answer]
I got it to work with 

```
import Stripe from "https://esm.sh/stripe@11.2.0?target=deno&no-check";
```

**Reply by thorwebdev** [reply]
Alternatively you can use an import map as mentioned here: https://github.com/stripe-samples/stripe-node-deno-samples/issues/2#issuecomment-1346642637

**Reply by galacticgibbon** [reply]
With this solution, I do not get typings (even the one that @thorwebdev mentioned which is specifically about lack of typings)

To make my life easier I'm just doing this for now 

```
import Stripe from "https://esm.sh/stripe@11.2.0?target=deno&no-check"
// import Stripe from "npm:stripe@11.2.0"

export const stripe = new Stripe(Deno.env.get("STRIPE_API_KEY") ?? '', {
  apiVersion: "2022-11-15",
});
```

I uncomment the npm module during development and then uncomment the esm version for deployment

All I can say is that deno annoys the hell out of me!

**Reply by thorwebdev** [reply]
@hamishjohnson can you try using an import_map like we do in the example here: 

* Import map: https://github.com/supabase/supabase/blob/master/examples/edge-functions/supabase/functions/import_map.json#L11
* Import in code: https://github.com/supabase/supabase/blob/master/examples/edge-functions/supabase/functions/stripe-webhooks/index.ts#L7-L8

With that types are working fine, see below:
<img width="1440" alt="image" src="https://github.com/supabase/supabase/assets/5748289/ad503b6b-5f76-4ba6-a620-929193878bf6">

**Reply by galacticgibbon** [reply]
Yeah, I did try with import map, and with deno.json also, but nothing worked

**Reply by rbkayz** [reply]
Same. It's very annoying
